fpm/src/fpm/git.f90

```diff
@@ -124,6 +128,45 @@ function git_target_tag(url, tag) result(self)
     end function git_target_tag
 
 
+    subroutine checkout(self,local_path, error)
+
+        !> Instance of the git target
+        class(git_target_t), intent(in) :: self
+
+        !> Local path to checkout in
+        character(*), intent(in) :: local_path
+
+        !> Error
+        type(error_t), allocatable, intent(out) :: error
+
+        !> Stat for execute_command_line
+        integer :: stat
+
+        call execute_command_line("git init "//local_path, exitstat=stat)
+
+        if (stat /= 0) then
+            call fatal_error(error,'Error while initiating git repository for remote dependency')
+            return
+        end if
+
+        call execute_command_line("git -C "//local_path//" fetch "//self%url//&
+                                        " "//self%object, exitstat=stat)
```

Comment:
There is no promise the git object field will be populated from the manifest. Either `git_target_default` sets the object to `HEAD` or you have to check here if an object was provided with the target.